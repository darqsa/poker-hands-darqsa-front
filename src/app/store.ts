import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import handsSlice from "../features/hands/slices/handsSlice";
import uiSlice from "../features/ui/slices/uiSlice";
import userSlice from "../features/users/slices/userSlice";

export const store = configureStore({
  reducer: {
    user: userSlice,
    hands: handsSlice,
    ui: uiSlice,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
