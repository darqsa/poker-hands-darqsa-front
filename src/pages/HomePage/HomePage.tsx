import Hand from "../../components/Hand/Hand";
import { HandData } from "../../features/hands/models/Hand";
import HomeContainerStyled from "./HomePageStyled";

const HomePage = (): JSX.Element => {
  const testHand: HandData = {
    handName: "Best hand name ever fpodsmfpomf",
    preGame: {
      hero: { hand: ["Ac", "Ad"], initialStack: 100, position: 0 },
      villains: [{ hand: ["Ah", "As"], initialStack: 100, position: 1 }],
    },
    game: {
      preFlop: { actions: ["Everyone is allin"], pot: 200 },
      flop: {
        board: ["Ts", "9c", "8h"],
        actions: ["Everyone is allin"],
        pot: 200,
      },
      turn: { board: "7d", actions: ["Everyone is allin"], pot: 200 },
      river: { board: "6d", actions: ["Everyone is allin"], pot: 200 },
    },
    postGame: { finalPot: 200, gameWinner: "Hero" },
    id: "1234",
  };

  return (
    <HomeContainerStyled className="home-container">
      {!testHand && (
        <p className="home-container__no-hands-text">
          You currently have no hands in your list... Try clicking at the
          top-left icon to create a new hand.
        </p>
      )}
      <Hand hand={testHand} />
    </HomeContainerStyled>
  );
};
export default HomePage;
