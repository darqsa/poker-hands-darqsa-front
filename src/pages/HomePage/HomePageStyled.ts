import styled from "styled-components";
import styles from "../../styles/styles";

const HomeContainerStyled = styled.div`
  display: flex;
  justify-content: center;
  color: ${styles.colors.main};
  .home-container {
    &__no-hands-text {
      font-weight: ${styles.fontWeights.bold};
      font-size: ${styles.fontSizes.bigger};
      text-align: center;
      margin-top: 100px;
      width: 80%;
      @media (min-width: 700px) {
        width: 35%;
      }
    }
  }
`;
export default HomeContainerStyled;
