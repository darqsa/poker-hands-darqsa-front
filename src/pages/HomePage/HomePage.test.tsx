import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { store } from "../../app/store";
import HomePage from "./HomePage";

describe("Given a home page component", () => {
  describe("When invoked", () => {
    test("Then it should match the snapshot", () => {
      const expectedHomePage = renderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <HomePage />
          </BrowserRouter>
        </Provider>
      );

      expect(expectedHomePage).toMatchSnapshot();
    });
  });
});
