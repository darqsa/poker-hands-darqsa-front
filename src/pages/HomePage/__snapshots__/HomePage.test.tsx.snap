// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a home page component When invoked Then it should match the snapshot 1`] = `
<div
  className="sc-gsnTZi QSfMP home-container"
>
  <article
    className="sc-bczRLJ fYErRj hand"
  >
    <div
      className="hand__hero-img-group"
    >
      <img
        alt="The Ac poker hand."
        className="hand__hero-card"
        height={75}
        src="./img/pokerCards/Ac.png"
      />
      <img
        alt="The Ad poker hand."
        className="hand__hero-card"
        height={75}
        src="./img/pokerCards/Ad.png"
      />
    </div>
    <div
      className="hand__hand-container"
    >
      <span
        className="hand__name"
      >
        Best hand name ever fpodsmfpomf
      </span>
      <div
        className="hand__board-img-group"
      >
        <img
          alt="The Ts poker hand."
          className="hand__board-card"
          height={60}
          src="./img/pokerCards/Ts.png"
        />
        <img
          alt="The 9c poker hand."
          className="hand__board-card"
          height={60}
          src="./img/pokerCards/9c.png"
        />
        <img
          alt="The 8h poker hand."
          className="hand__board-card"
          height={60}
          src="./img/pokerCards/8h.png"
        />
        <img
          alt="The 7d poker hand."
          className="hand__board-card"
          height={60}
          src="./img/pokerCards/7d.png"
        />
        <img
          alt="The 6d poker hand."
          className="hand__board-card"
          height={60}
          src="./img/pokerCards/6d.png"
        />
      </div>
    </div>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium hand__more-icon css-i4bv87-MuiSvgIcon-root"
      data-testid="more-vert"
      focusable="false"
      onClick={[Function]}
      viewBox="0 0 24 24"
    >
      <path
        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
      />
    </svg>
  </article>
</div>
`;
